File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
// 3. 여러 줄의 값들을 입력받을 때
const [N, ...input] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");
/*
마감시간 기준 오름차순 정렬
*/
let stack = [];
let arr = input.map((e) => {
  return e.split(" ").map(Number);
});
arr.sort((a, b) => {
  if (a[1] == b[1]) return a[0] - b[0];
  else return a[1] - b[1];
});

stack.push(arr[0][1]);

for (let i = 1; i < arr.length; i++) {
  if (arr[i][0] >= stack[stack.length - 1]) {
    stack.push(arr[i][1]);
  }
}
console.log(stack.length);

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys
import copy
from collections import deque
input = sys.stdin.readline
#아이디어
# 그리디, 종료시간 기준 오름차순 정렬
# while문으로 포인터 하나 설정해서 currEnd보다 작으면 pass, 크면 그거 넣기
# 시간복잡도
# 선형시간
# 변수 사용 계획
# count , i
# n = int(input()) # 계단 갯수
n = int(input())
meetings = []
for _ in range(n):
    meetings.append(list(map(int,input().split())))
meetings.sort(key=lambda x : (x[1],x[0]))

def solution (meetings,n) :
    ans = []
    for x in meetings:
        if not ans :
            ans.append(x)
        else :
            if ans[-1][1] <= x[0]:
                ans.append(x)
    return len(ans)
print(solution(meetings,n))